src/test/java/unit/AiNotebookControlled.java

```diff
@@ -0,0 +1,312 @@
+
+    // `terra resources describe --name=$name --format=json`
+    UFAiNotebook describeResource =
+        TestCommand.runAndParseCommandExpectSuccess(
+            UFAiNotebook.class, "resources", "describe", "--name=" + name);
+
+    // check that the properties match
+    assertEquals(name, describeResource.name, "describe resource output matches name");
+    assertEquals(cloning, describeResource.cloningInstructions, "describe output matches cloning");
+    assertEquals(description, describeResource.description, "describe output matches description");
+    assertEquals(location, describeResource.location, "describe resource output matches location");
+    assertEquals(
+        instanceId, describeResource.instanceId, "describe resource output matches instance id");
+
+    // ai notebooks are always private
+    assertEquals(
+        AccessScope.PRIVATE_ACCESS, describeResource.accessScope, "describe output matches access");
+    assertEquals(
+        workspaceCreator.email.toLowerCase(),
+        describeResource.privateUserName.toLowerCase(),
+        "describe output matches private user name");
+    // TODO (PF-616): check the private user roles once WSM returns them
+  }
+
+  @Test // NOTE: This test takes ~10 minutes to run.
+  @DisplayName("start, stop a notebook and poll until they complete")
+  void startStop() throws IOException, InterruptedException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra resources create ai-notebook --name=$name`
+    String name = "startStop";
+    TestCommand.runCommandExpectSuccess("resources", "create", "ai-notebook", "--name=" + name);
+    pollDescribeForNotebookState(name, "PROVISIONING");
+
+    // `terra notebooks start --name=$name`
+    // TODO (PF-869): change this to expect success once polling notebook operations is fixed
+    TestCommand.runCommand("notebooks", "start", "--name=" + name);
+    pollDescribeForNotebookState(name, "ACTIVE");
+
+    // `terra notebooks stop --name=$name`
+    // TODO (PF-869): change this to expect success once polling notebook operations is fixed
+    TestCommand.runCommand("notebooks", "stop", "--name=" + name);
+    pollDescribeForNotebookState(name, "STOPPED");
+  }
+
+  /**
+   * Helper method to poll `terra resources describe` until the notebook state equals that
+   * specified. Uses the current workspace.
+   */
+  public static void pollDescribeForNotebookState(String resourceName, String notebookState)
+      throws InterruptedException, JsonProcessingException {
+    pollDescribeForNotebookState(resourceName, notebookState, null);
+  }
+
+  /**
+   * Helper method to poll `terra resources describe` until the notebook state equals that
+   * specified. Filters on the specified workspace id; Uses the current workspace if null.
+   */
+  public static void pollDescribeForNotebookState(
```

Comment:
Why poll through a CLI command instead of using `GoogleAiNotebooks.get`? Would be less verbose and more to the point.

Reply:
I was trying to avoid testing WSM functionality, and only test the CLI. That's not always possible (e.g. for bucket lifecycle rules that WSM doesn't store, I have to query the cloud directly), but where it is, I wanted to take advantage and exercise the CLI more. If we ever add tests for the other notebook creation parameters, I think we will have to query the cloud directly.